scripts/query_mlwh.py

```diff
@@ -0,0 +1,152 @@
+    
+    Return pandas dataframe of sample metadata
+    '''
+    
+    # Open database connection
+    db = MySQLdb.connect(read_default_file=db_config_file)
+    
+    # prepare a cursor object using cursor() method
+    cursor = db.cursor()
+    
+    try:
+        
+        # This query will give us all the samples that have been sequenced
+        # for sample identifiers from a list
+        base_query = """
+             SELECT 
+                sample.name, 
+                sample.supplier_name,
+                sample.accession_number,
+                study.name,
+                iseq_flowcell.flowcell_barcode,
+                iseq_flowcell.legacy_library_id,
+                iseq_flowcell.manual_qc,
+                iseq_run_lane_metrics.run_complete,
+                iseq_run_lane_metrics.qc_complete,
+                iseq_product_metrics.id_run,
+                iseq_product_metrics.position,
+                iseq_flowcell.tag_index
+               
+             FROM 
+                sample, 
+                iseq_flowcell, 
+                study, 
+                iseq_product_metrics, 
+                iseq_run_lane_metrics
+       
+             WHERE 
+                study.id_study_tmp = iseq_flowcell.id_study_tmp
+                and iseq_flowcell.id_sample_tmp = sample.id_sample_tmp
+                and iseq_product_metrics.id_iseq_flowcell_tmp = iseq_flowcell.id_iseq_flowcell_tmp
+                and iseq_product_metrics.tag_index = iseq_flowcell.tag_index
+                and iseq_product_metrics.position = iseq_flowcell.position
+                and iseq_run_lane_metrics.flowcell_barcode = iseq_flowcell.flowcell_barcode
+                and iseq_run_lane_metrics.id_run = iseq_product_metrics.id_run
+                and iseq_run_lane_metrics.position = iseq_product_metrics.position
+                and sample.{id_type} in ("{sample_list}")
+                """
+        query = base_query.format(id_type=id_type, 
+                                  sample_list='", "'.join(sample_list))
+        cursor.execute(query)
+
+        # Fetch tuple of tuples.  Each inner tuple is a row.
+        db_results = cursor.fetchall()
+        
+        sql_data = pd.DataFrame.from_records(
+            list(db_results),
+            columns=["sample_name", 
+                     "sample_supplier_name", 
+                     "ena_sample_accession",
+                     "study", 
+                     "flowcell_barcode", 
+                     "library",
+                     "manual_qc", 
+                     "run_complete", 
+                     "qc_complete",
+                     "run_id", 
+                     "lane_index", 
+                     "tag_index"])
+        
+        # include empty ebi run accession column (data not available from mlwh)
+        sql_data["ena_run_accession"] = ""
+        
+        return (sql_data)
+
+    finally:           
+        if db and db.open:
```

Comment:
Small point, but this check seems unnecessary. I.e., you can't get to this point unless `db` is an open connection. 

Also usually one calls `cursor.close()` before `db.close()`, but maybe that's not needed as `db.close()` will handle all cleanup.